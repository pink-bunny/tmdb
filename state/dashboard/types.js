export const FETCH_TRENDING_MOVIES = 'FETCH_TRENDING_MOVIES';
export const FETCH_TRENDING_MOVIES_SUCCESS = 'FETCH_TRENDING_MOVIES_SUCCESS';
export const FETCH_TRENDING_MOVIES_ERROR = 'FETCH_TRENDING_MOVIES_ERROR';
