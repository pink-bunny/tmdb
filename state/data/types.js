export const MERGE_MOVIES_LIST = 'MERGE_MOVIES_LIST';

export const MERGE_LISTS_LIST = 'MERGE_LISTS_LIST';
