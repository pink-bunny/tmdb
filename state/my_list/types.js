export const FETCH_MY_LIST = 'FETCH_MY_LIST';
export const FETCH_MY_LIST_SUCCESS = 'FETCH_MY_LIST_SUCCESS';
export const FETCH_MY_LIST_ERROR = 'FETCH_MY_LIST_ERROR';

export const CREATE_MY_LIST = 'CREATE_MY_LIST';

export const DELETE_MY_LIST = 'DELETE_MY_LIST';
export const DELETE_MY_LIST_ERROR = 'DELETE_MY_LIST_ERROR';

export const TOGGLE_TO_LIST = 'TOGGLE_TO_LIST';
export const TOGGLE_TO_LIST_SUCCESS = 'TOGGLE_TO_LIST_SUCCESS';

export const CREATE_AND_ADD_TO_LIST = 'CREATE_AND_ADD_TO_LIST';
