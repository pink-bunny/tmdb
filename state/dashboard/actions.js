import * as types from './types';

export const fetchTrendingMovies = (page = 1) => ({
  type: types.FETCH_TRENDING_MOVIES,
  page
});

export const fetchTrendingMoviesSuccess = (ids, totalItems, currentPage) => ({
  type: types.FETCH_TRENDING_MOVIES_SUCCESS,
  ids,
  totalItems,
  currentPage
});

export const fetchTrendingMoviesError = (error) => ({
  type: types.FETCH_TRENDING_MOVIES_ERROR,
  error
});

export const searchMovies = ({ search }, setErrors, setSubmitting, page = 1) => ({
  type: types.SEARCH_MOVIES,
  search,
  page,
  setErrors,
  setSubmitting
});
