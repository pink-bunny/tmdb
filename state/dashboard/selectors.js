export const isDashboardMoviesLoading = (state) => state.dashboard.loading;

export const dashboardMoviesList = (state) => {
  const refs = state.dashboard.ids;
  const list = state.data.movies;

  if (refs.length === 0) {
    return null;
  }
  return refs.map((item) => list[item]);
};

export const dashboardMoviesTotalPages = (state) => (
  state.dashboard.totalItems
);

export const dashboardMoviesCurrentPage = (state) => (
  state.dashboard.currentPage
);

export const dashboardMoviesError = (state) => state.dashboard.error;

export const dashboardMoviesSearchQuery = (state) => state.dashboard.searchQuery;
