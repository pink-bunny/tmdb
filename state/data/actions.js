import * as types from './types';
import { TOGGLE_TO_WATCHLIST_SUCCESS } from '../movie/types';

export const mergeMoviesList = (movies) => ({
  type: types.MERGE_MOVIES_LIST,
  movies
});

export const mergeListsList = (lists) => ({
  type: types.MERGE_LISTS_LIST,
  lists
});

export const toggleToWatchlistSuccess = (id, isMovieInWatchlist) => ({
  type: TOGGLE_TO_WATCHLIST_SUCCESS,
  id,
  isMovieInWatchlist
});
