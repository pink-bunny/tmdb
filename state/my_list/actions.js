import * as types from './types';

export const fetchMyList = (id) => ({
  type: types.FETCH_MY_LIST,
  id
});

export const fetchMyListSuccess = (movieId) => ({
  type: types.FETCH_MY_LIST_SUCCESS,
  movieId
});

export const fetchMyListError = (error) => ({
  type: types.FETCH_MY_LIST_ERROR,
  error
});
