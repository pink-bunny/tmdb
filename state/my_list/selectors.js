export const isMyListLoading = (state) => state.myList.loading;

export const myListError = (state) => state.myList.error;

export const myListInfo = (state) => {
  const id = state.myList.movieId || null;
  const list = state.data.lists || {};

  if (!id) {
    return null;
  }
  return list[id];
};

export const deleteMyListError = (state) => state.myLists.deleteMyListError;
