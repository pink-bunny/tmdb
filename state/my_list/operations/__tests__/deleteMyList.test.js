import deleteMyListLogic from '../deleteMyList';
import { DELETE_MY_LIST } from '../../types';
import mockHttpClient from '../../../../utils/testsHelpers/mockHttpClient';
import { getSessionId } from '../../../session/selectors';
import { fetchMyLists } from '../../../my_lists/actions';
import { deleteMyListError } from '../../actions';
import deleteMyListResponseError from '../__mocks__/deleteMyList';

jest.mock('../../actions', () => ({
  deleteMyListError: jest.fn()
}));
jest.mock('../../../my_lists/actions', () => ({
  fetchMyLists: jest.fn()
}));

describe('deleteMyList logic', () => {
  const hideModal = jest.fn();
  const action = {
    type: DELETE_MY_LIST,
    id: 1,
    hideModal
  };
  const { id } = action;
  const dispatch = jest.fn();
  const done = jest.fn();
  const getState = jest.fn(() => ({ session: { sessionId: '1q2w3e4r5t' } }));
  const state = getState();
  const sessionId = getSessionId(state);
  let httpClient;
  httpClient = mockHttpClient({ method: 'delete' });
  deleteMyListLogic.process({ getState, httpClient, action }, dispatch, done);

  it('matches snapshot', () => {
    expect(deleteMyListLogic).toMatchSnapshot();
  });

  it('called with right arguments', () => {
    const requestPath = `list/${id}?session_id=${sessionId}`;
    expect(httpClient.delete).toHaveBeenCalledWith(requestPath);
  });

  describe('success.', () => {
    it('Dispatch fetchMyLists()', () => {
      expect(fetchMyLists).toHaveBeenCalled();
    });
    it('Hide modal', () => {
      expect(hideModal).toHaveBeenCalled();
    });
  });

  describe('failure.', () => {
    httpClient = mockHttpClient({ method: 'delete', response: deleteMyListResponseError, reject: true });
    deleteMyListLogic.process({ getState, httpClient, action }, dispatch, done);

    it('Dispatch deleteMyListError() with error', () => {
      const error = deleteMyListResponseError;
      const errorMessage = error.response.data.status_message;
      expect(deleteMyListError).toHaveBeenCalledWith(errorMessage);
    });
    it('Dispatch fetchMyLists()', () => {
      expect(fetchMyLists).toHaveBeenCalled();
    });
  });

  it('done() have been called', () => {
    expect(done).toHaveBeenCalled();
  });
});
