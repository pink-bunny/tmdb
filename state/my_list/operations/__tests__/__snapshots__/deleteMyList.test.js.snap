// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deleteMyList logic matches snapshot 1`] = `
Object {
  "cancelType": undefined,
  "debounce": 0,
  "latest": false,
  "name": undefined,
  "process": [Function],
  "processOptions": Object {
    "dispatchMultiple": true,
  },
  "throttle": 0,
  "transform": undefined,
  "type": "DELETE_MY_LIST",
  "validate": undefined,
  "warnTimeout": 60000,
}
`;
